pkg/devfile/parser/data/v2/header.go

```diff
@@ -0,0 +1,23 @@
+package v2
+
+import (
+	devfilepkg "github.com/devfile/api/pkg/devfile"
+)
+
+//SetSchemaVersion sets devfile schema version
+func (d *DevfileV2) SetSchemaVersion(version string) {
+	d.SchemaVersion = version
+}
+
+// GetMetadata returns the DevfileMetadata Object parsed from devfile
+func (d *DevfileV2) GetMetadata() devfilepkg.DevfileMetadata {
+	return d.DevfileHeader.Metadata
```

Comment:
While there are benefits to resolving the inlined struct in this way, it's probably not necessary -- we can just do
```
return d.Metadata
```
at which point this function seems unnecessary.